airflow/providers/google/cloud/hooks/vertex_ai/custom_job.py

```diff
@@ -0,0 +1,2379 @@
+        :param pipeline_job_id:  The ID to use for the PipelineJob, which will become the final component of
+            the PipelineJob name. If not provided, an ID will be automatically generated.
+
+            This value should be less than 128 characters, and valid characters are /[a-z][0-9]-/.
+        :type pipeline_job_id: str
+        :param retry: Designation of what errors, if any, should be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The timeout for this request.
+        :type timeout: float
+        :param metadata: Strings which should be sent along with the request as metadata.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_pipeline_service_client(region)
+        parent = client.common_location_path(project_id, region)
+
+        result = client.create_pipeline_job(
+            request={
+                'parent': parent,
+                'pipeline_job': pipeline_job,
+                'pipeline_job_id': pipeline_job_id,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+        return result
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def create_training_pipeline(
+        self,
+        project_id: str,
+        region: str,
+        training_pipeline: TrainingPipeline,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+    ) -> TrainingPipeline:
+        """
+        Creates a TrainingPipeline. A created TrainingPipeline right away will be attempted to be run.
+
+        :param project_id: Required. The ID of the Google Cloud project that the service belongs to.
+        :type project_id: str
+        :param region: Required. The ID of the Google Cloud region that the service belongs to.
+        :type region: str
+        :param training_pipeline:  Required. The TrainingPipeline to create.
+        :type training_pipeline: google.cloud.aiplatform_v1.types.TrainingPipeline
+        :param retry: Designation of what errors, if any, should be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The timeout for this request.
+        :type timeout: float
+        :param metadata: Strings which should be sent along with the request as metadata.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_pipeline_service_client(region)
+        parent = client.common_location_path(project_id, region)
+
+        result = client.create_training_pipeline(
+            request={
+                'parent': parent,
+                'training_pipeline': training_pipeline,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+        return result
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def create_custom_job(
```

Comment:
Do we need this? I don't see that we using it.

Reply:
I have answered to the same question in message above

Reply:
👍